ADBench/utils.py

```diff
@@ -59,10 +59,13 @@ def _mkdir_if_none(path):
 def cap_str(s):
     return s[0].upper() + (s[1:] if len(s) > 1 else "")
 
+# Remove extension from the file name
+def get_no_ext(file_name):
+    return file_name.split(".")[-2]
```

Comment:
@mikhailnikolaev I'm confused why this uses `-2` unlike the original, which used `0`.  They will be the same if and only if the split path has two entries. Are there cases where we expect this to not be the case?

For maximal robustness perhaps we should literally strip off *only* the extension part, i.e. `get_no_ext("a.b.ext")` should be `"a.b"`.  There's probably a Python function to do this?  What do you think?

Reply:
Oh, it seems that I was wrong in here.
I've writen -2 exactly for case when a filename is something like this "a.b.c.txt". Then an extension is only ".txt", but not ".b.c.txt". As you can see the old code, using `0` would recognize an extension as in the second case. But for some reason I didn't recognize that new code will return exactly wrong result.

Yes, I think your idea is right. I've just found a right way using a function from `os.path`:
```Python
return os.path.splitext(file_name)[0]
```
This will work exactly correct and, e.g. will return "a.b.c" if the given name is "a.b.c.txt"

So, will I do such a patch or you will do this yourself?

Reply:
Your proposal looks good.  Please push it as an additional commit to the PR.  I still have to find the time to look at the rest of the PR in depth ...

Reply:
I also suggest removing `get_no_ext` and `get_fn` entirely and replacing both uses of `get_fn` in `plot_graphs.py` (and elsewhere, if any) with `os.path.splitext`.

Reply:
Done